packages/core/src/external/cda/process-attachments.ts

```diff
@@ -0,0 +1,259 @@
+import { cwExtension } from "../commonwell/extension";
+import { makeFhirApi } from "../fhir/api/api-factory";
+import { convertCollectionBundleToTransactionBundle } from "../fhir/bundle/convert-to-transaction-bundle";
+import { buildDocIdFhirExtension } from "../fhir/shared/extensions/doc-id-extension";
+
+const region = Config.getAWSRegion();
+
+function getS3UtilsInstance(): S3Utils {
+  return new S3Utils(region);
+}
+export async function processAttachments({
+  b64Attachments,
+  cxId,
+  patientId,
+  filePath,
+  medicalDataSource,
+  s3BucketName,
+  fhirUrl,
+}: {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  b64Attachments: any[];
+  cxId: string;
+  patientId: string;
+  filePath: string;
+  medicalDataSource?: string | undefined;
+  s3BucketName: string | undefined;
+  fhirUrl?: string;
+}) {
+  const { log } = out(`processAttachments - cxId ${cxId}, patientId ${patientId}`);
+  const s3Utils = getS3UtilsInstance();
+
+  const extensions = [buildDocIdFhirExtension(filePath), getSourceExtension(medicalDataSource)]
+    .flat()
+    .filter(Boolean) as Extension[];
+
+  const docRefs: DocumentReference[] = [];
+  const uploadDetails: UploadParams[] = [];
+
+  b64Attachments.map(async att => {
+    const act = att.act as ConcernActEntryAct;
+
+    const fileDetails = getDetails(act.text);
+    if (!fileDetails.fileB64Contents) return;
+
+    const docRef = buildDocumentReference(patientId, extensions, act);
+    if (!docRef.id) throw new Error("Missing ID in DocRef");
+
+    const fileKey = createAttachmentUploadFilePath({
+      filePath,
+      attachmentId: docRef.id,
+      mimeType: fileDetails.mimeType,
+    });
+
+    const fileUrl = s3Utils.buildFileUrl(s3BucketName ?? "", fileKey);
+
+    const attachment: Attachment = {
+      ...(fileDetails.mimeType && { contentType: fileDetails.mimeType }),
+      url: fileUrl,
+      size: sizeInBytes(fileDetails.fileB64Contents),
+      title: fileKey,
+    };
+
+    if (docRef.date) attachment.creation = docRef.date;
+    docRef.content = [{ attachment }];
+
+    docRefs.push(docRef);
+
+    const uploadParams = {
+      bucket: s3BucketName ?? "",
+      key: fileKey,
+      file: Buffer.from(fileDetails.fileB64Contents, "base64"),
+      ...(fileDetails.mimeType && { contentType: fileDetails.mimeType }),
+    };
+    uploadDetails.push(uploadParams);
+  });
+
+  log(`Extracted ${docRefs.length} attachments`);
+  const docRefBundleEntries = docRefs.map(dr => ({ resource: dr }));
+  const collectionBundle: Bundle = {
+    resourceType: "Bundle",
+    type: "collection",
+    entry: docRefBundleEntries,
+  };
+
+  const transactionBundle = convertCollectionBundleToTransactionBundle({
+    fhirBundle: collectionBundle,
+  });
+
+  if (transactionBundle.entry?.length) {
+    await Promise.all([
+      handleFhirUpload(transactionBundle, cxId, log, fhirUrl),
+      handleS3Upload(uploadDetails, log, s3Utils, s3BucketName),
```

Comment:
needs to be `s3BucketName ? handleS3Upload(uploadDetails, log, s3Utils, s3BucketName) : undefined` or something -- again, so reason to do anything s3 related if the bucket is not defined.